flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -586,19 +606,20 @@ private ChangelogSnapshotState completeRestore(
 
             LOG.info("Starting materialization from {} : {}", lastMaterializedTo, upTo);
 
+            // This ID is not needed for materialization; But since we are re-using the
+            // streamFactory that is designed for state backend snapshot, which requires unique
+            // checkpoint ID. A faked materialized Id is provided here.
+            // TODO: implement its own streamFactory.
```

Comment:
I think this TODO for a stream factory should be commented on the `streamFactory`.